sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -420,4 +420,14 @@ class UDFSuite extends QueryTest with SharedSQLContext {
       checkAnswer(df, Seq(Row("null1x"), Row(null), Row("N3null")))
     }
   }
+
+  test("SPARK-26308: udf with decimal") {
+    val df1 = spark.createDataFrame(
+      sparkContext.parallelize(Seq(Row(new java.math.BigDecimal("2011000000000002456556")))),
+      StructType(Seq(StructField("col1", DecimalType(30, 0)))))
+    val udf1 = org.apache.spark.sql.functions.udf((value: java.math.BigDecimal) => {
+      if (value == null) null else value.toBigInteger.toString
```

Comment:
before this fix, the value passed here is null. This happens because the decimal value is casted to `Decimal(38, 18)` which is not enough to contain that value: so it becomes null. I preferred the test to fail for a bad result rather than a NPE, without this fix.